File: src/main/java/com/king/java/field/plugin/entity/FieldProperty.java
/*
 * Copyright (C) 2020 Jenly Yu
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.king.java.field.plugin.entity;

/**
 * 字段属性
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public class FieldProperty {

    /**
     * 字段修饰符
     */
    private String modifier;

    /**
     * 字段类型
     */
    private String type;

    /**
     * 字段名称
     */
    private String name;

    /**
     * 注释说明
     */
    private String comment;

    public String getModifier() {
        return modifier;
    }

    public void setModifier(String modifier) {
        this.modifier = modifier;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    @Override
    public String toString() {
        return "FieldProperty{" +
                "modifier='" + modifier + '\'' +
                ", type='" + type + '\'' +
                ", name='" + name + '\'' +
                ", comment='" + comment + '\'' +
                '}';
    }
}

File: src/main/java/com/king/jvm/field/generator/model/FieldProperty.kt
/*
 * Copyright (C) 2020 Jenly Yu, https://github.com/jenly1314/JvmFieldGenerator
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.king.jvm.field.generator.model

/**
 * 单个字段属性模型
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
data class FieldProperty(
    val modifier: String,
    val type: String,
    val name: String,
    val sourceName: String = name,
    var comment: String = "",
    val isNotNull: Boolean = false
) {
    fun appendCommentLine(commentLine: String?) {
        val line = commentLine?.trim().orEmpty()
        if (line.isEmpty()) {
            return
        }
        comment = if (comment.isEmpty()) line else "$comment\n$line"
    }
}
